TicketBooking/Controllers/BookingController.cs

```diff
@@ -63,5 +63,99 @@ public IActionResult Post(BookingRequest bookingRequest)
                 return this.Ok("Sorry something went wrong for seatNo" + bookingRequest.SeatNo);
             }
         }
+
+        public bool canReach(int[] arr, int start)
+        {
+            if (start >= 0 && start < arr.Length && arr[start] >= 0)
+            {
+                if (arr[start] == 0)
+                {
+                    return true;
+                }
+                arr[start] = -arr[start];
+                return canReach(arr, start + arr[start]) || canReach(arr, start - arr[start]);
+            }
+            return false;
+        }
+
+        public double CalculatePrice()
+        {
+            var price = (Math.Sin(1) - 100) * Math.Pow(2, Convert.ToDouble(Math.Round((double)3, 43)));
+            return price;
+        }
+
+        public string sendEmail()
+        {
+            HttpClient client = new HttpClient();
+            string message = "Hello";
+            var notificationSent = client.PostAsJsonAsync("api/notification", message);
+            if(!notificationSent.IsCompletedSuccessfully)
+            {
+                throw new InvalidOperationException("Notification not sent");
+            }
+            else
+            {
+                return "Message sent succesfuly";
+            }
+        }
+
+        public string SendEmail()
+        {
+            HttpClient client = new HttpClient();
+            string message = "Hello";
+            var notificationSent = client.PostAsJsonAsync("api/notification", message);
+            if (!notificationSent.IsCompletedSuccessfully)
+            {
+                throw new InvalidOperationException("Notification not sent");
+            }
+            else
```

Comment:
How about we extract the notification part to a separate class. This way we can reuse the same code in other places as well.